src/styles/main.scss

```diff
@@ -0,0 +1,65 @@
+$day-size: 100px;
+$day-gap: 1px;
+$calendar-padding: 10px;
+$calendar-column: 7;
+$calendar-width: $day-size * $calendar-column + $day-gap * ($calendar-column - 1) + 2 * $calendar-padding;
+$start-day: ('mon' 0, 'tue' 1, 'wed' 2, 'thu' 3, 'fri' 4, 'sat' 5, 'sun' 6);
+
+body {
+  margin: 0;
+  font-family: Arial, Helvetica, sans-serif;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  box-sizing: border-box;
+  display: flex;
+  padding: $calendar-padding;
+  width: $calendar-width;
+  gap: $day-gap;
+  flex-wrap: wrap;
+
+  &__day {
+    display: flex;
+    box-sizing: border-box;
+    width: $day-size;
+    height: $day-size;
+    background-color: #eee;
+    border: solid 1px #000;
+    position: relative;
+    transition: 0.5s;
+    
+    &:hover {
+      cursor: pointer;
+      transform: translateY(-20px);
+      background-color: #ffbfcb;
+    }
+  }
+
+  @for $day from 1 through 31 {
+    &__day:nth-child(#{$day})::before {
+      content: '#{$day}';
+      font-size: 30px;
+      position: absolute;
+      color: black;
+      left: 50%;
+      top: 50%;
+      transform: translate(-50%, -50%);
```

Comment:
The `&__day:nth-child(#{$day})::before` pseudo-element (lines 43-50) uses absolute positioning to center its content. While the current approach works, consider using flexbox for centering as it's more in line with modern CSS practices and doesn't require translation adjustments.